test/spec/Control/Biegunka/Source/GitSpec.hs

```diff
@@ -0,0 +1,82 @@
+module Control.Biegunka.Source.GitSpec (spec) where
+import           Test.Hspec
+
+import           Control.Biegunka
+import           Control.Biegunka.Source.Git (git_)
+import           Control.Lens
+import           Data.Maybe                  (listToMaybe)
+import           System.FilePath             ((</>))
+import           System.IO.Temp              (withSystemTempDirectory)
+import qualified System.Process              as P
+
+spec :: Spec
+spec = do
+  describe "git_" $ do
+
+    context "when local path doesn't exist" $
+      it "creates new directory and set branch correctly" $
+        withBiegunkaDirectory $ \tmp -> do
+          let repo = tmp </> "biegunka-test"
+          biegunka (set runRoot tmp . set biegunkaRoot (tmp </> ".biegunka")) run $ -- TODO: suppress biegunka's output
+            git_ testRepo "biegunka-test"
+          currentBranch repo `shouldReturn` Just "master"
+          repoHaveCleanState repo `shouldReturn` True
+
+    context "when local branch have no commits ahead of remote branch and remote branch have no new commits" $
```

Comment:
You can avoid repeating "when local branch..."  if you extract it into a separate context (they nest).